### DISCUSSION TITLE
cors block on localhost dev

### DISCUSSION THREAD

**Discussion by lilpaul** [WalletConnect/walletconnect-monorepo#4456] [Developer Support]
Hey guys
I set up an account to try integrate the web3modal into our dapp, but I am getting cors errors when running the app locally for dev. The documentation all says localhost should always be enabled, so I'm not sure what I can do from here. I have verified the project via txt dns record, and tried it with and without our domain set in the allowlist.
Not sure if its because I am running it on https://localhost and maybe only http is allowed for localhost or something like that maybe, I have also tried it on various ports to check if it was the custom port I was using, but it seems blocked on 80 and other ports.
The account the project is in, is devops@theartbank.io
Please let me know if there is any other info I can provide or if there is something else I can configure in the project setup to allow it. I have tried it with the verify api enabled and disabled.
Thanks.
Paul

**Comment by MuckT** [comment]
I'd expect `https://localhost:<port>` to require a valid certificate to work. If `http://localhost:<port>` isn't working try `127.0.0.1:<port>`.

**Reply by lilpaul** [reply]
thanks for the input, I hadn't thought of it maybe being a cert issue, so I tried making a new cert using mkcert with no luck, and made a cert for dev.{myproject.tld} ran the project with that mapped in my hosts file and added that as an allowed domain with wildcard and with dev. in the project settings with no luck.
Also deployed a test build to our uat site at uat.{myproject.tld} which has an azure managed cert and still got cors blocked on there, also using wildcard and uat. in the allowlist.
I am using this nuget package component (https://www.nuget.org/packages/Blazor-Web3Modal), and have configured the project_Id according to their documentation. I can see the component on my page but obviously it can't do anything with the cors block.

**Comment by Cali93** [comment]
Hey @lilpaul,

Sorry for the delay and thanks for reporting this issue. 
Could you please share the exact error you're getting ?
